python/tank/bootstrap/manager.py

```diff
@@ -354,13 +354,43 @@ def bootstrap_engine(self, engine_name, entity=None):
         self._log_startup_message(engine_name, entity)
 
         tk = self._bootstrap_sgtk(engine_name, entity)
-
         engine = self._start_engine(tk, engine_name, entity)
 
         self._report_progress(self.progress_callback, self._BOOTSTRAP_COMPLETED, "Engine launched.")
 
         return engine
 
+    def bootstrap_legacy_shotgun_engine(self, entity=None):
+        """
+        Create an :class:`~sgtk.Sgtk` instance for the tk-shotgun engine and entity,
+        then launch into the tk-shotgun engine making use of a legacy shotgun_*.yml environment.
+
+        The whole engine bootstrap logic will be executed synchronously in the main application thread.
+
+        If entity is None, the method will bootstrap into the site
+        config. This method will attempt to resolve the config according
+        to business logic set in the associated resolver class and based
+        on this launch a configuration. This may involve downloading new
+        apps from the toolkit app store and installing files on disk.
+
+        Please note that the API version of the tk instance that hosts
+        the engine may not be the same as the API version that was
+        executed during the bootstrap.
+
+        :param entity: Shotgun entity to launch engine for.
+        :type entity: Dictionary with keys ``type`` and ``id``, or ``None`` for the site.
+        :returns: :class:`~sgtk.platform.Engine` instance.
+        """
+        engine_name = "tk-shotgun"
```

Comment:
in a comment, i think we should point out that this is the "bootstrap equivalent" of the `tank` command entry point here: 

- https://github.com/shotgunsoftware/tk-core/blob/master/scripts/tank_cmd.py#L301
- https://github.com/shotgunsoftware/tk-core/blob/master/scripts/tank_cmd.py#L351